frontend/cypress/e2e/Pages/User/UserCreation.test.js

```diff
@@ -0,0 +1,88 @@
+import {
+  Given,
+  When,
+  Then,
+  After
+} from '@badeball/cypress-cucumber-preprocessor'
+
+Given('the IDIR user is on the login page', () => {
+  cy.clearAllCookies()
+  cy.clearAllLocalStorage()
+  cy.clearAllSessionStorage()
+  cy.visit('/')
+  cy.getByDataTest('login-container').should('exist')
+})
+
+When('the IDIR user logs in with valid credentials', () => {
+  cy.loginWith(
+    'idir',
+    Cypress.env('admin_idir_username'),
+    Cypress.env('admin_idir_password')
+  )
+  cy.wait(5000)
+  cy.setIDIRRoles('analyst')
+  cy.visit('/')
+  cy.getByDataTest('dashboard-container').should('exist')
+})
+
+When('the IDIR user navigates to the user creation page', () => {
+    cy.get('a[href="/admin"]').click()
+    cy.wait(3000)
+    cy.url().should('include', '/admin/users')
+    cy.contains('New user').click()
+    cy.url().should('include', '/admin/users/add-user')
+})
+
+When('the IDIR user fills out the form with valid data', () => {
+    cy.get('input[id="firstName"]').type('John')
+    cy.get('input[id="lastName"]').type('Doe')
+    cy.get('input[id="jobTitle"]').type('Senior Analyst')
+    cy.get('input[id="userName"]').type('johndoe')
+    cy.get('input[id="keycloakEmail"]').type('john.doe@example.com')
+    cy.get('input[id="phone"]').type('1234567890')
+    cy.get('input[id="mobilePhone"]').type('0987654321')
+
+    // Select the Analyst role
+    cy.get('input[type="radio"][value="analyst"]').check()
+})
+
+When('the IDIR user submits the form', () => {
+    cy.get('button[data-test="saveUser"]').click()
+    cy.wait(3000)
```

Comment:
Same here, is there an element or url change we can wait for instead?

Reply:
Nice idea!